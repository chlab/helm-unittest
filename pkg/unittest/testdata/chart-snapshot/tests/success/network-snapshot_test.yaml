suite: test ingress
templates:
  - templates/network.yaml

tests:
  - it: should render snapshot
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}
      - matchSnapshot:
          path: metadata.labels
      - matchSnapshot:
          matchRegex:
            pattern: .*app.*
      - matchSnapshot:
          notMatchRegex:
            pattern: .*app\.kubernetes\.io/version-wrong.*

  - it: should render snapshot and assert in single matcher
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot:
          path: metadata.labels
          matchRegex:
            pattern: .*app.*
          notMatchRegex:
            pattern: .*app\.kubernetes\.io/version-wrong.*
