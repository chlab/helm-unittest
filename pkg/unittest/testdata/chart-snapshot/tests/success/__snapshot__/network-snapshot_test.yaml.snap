should render snapshot:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app: test-cluster
        app.kubernetes.io/version: null
  2: |
    app: test-cluster
    app.kubernetes.io/version: null
  3: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app: test-cluster
        app.kubernetes.io/version: null
  4: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      labels:
        app: test-cluster
        app.kubernetes.io/version: null
should render snapshot and assert in single matcher:
  1: |
    app: test-cluster
    app.kubernetes.io/version: null
